File: KarmaAndroid/app/src/main/java/edu/upenn/cis350/karma/Vendor.java
package edu.upenn.cis350.karma;

public class Vendor {

    private String name;
    private String location;
    private String description;
    private String email;

    public Vendor(String name, String location, String description, String email) {
        this.name = name;
        this.location = location;
        this.description = description;
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public String getLocation() {
        return location;
    }

    public String getEmail() {return email;};

    public String getDescription() {
        return description;
    }
}

File: KarmaWeb/models/Vendor.js
//Set up mongoose connection
var mongoose = require('mongoose');
var mongoDB = 'mongodb+srv://xavlee:karmadb@karma-4qqqe.mongodb.net/Karma';
mongoose.connect(mongoDB, { useNewUrlParser: true });
var db = mongoose.connection;
db.on('error', console.error.bind(console, 'MongoDB connection error:'));

var Schema = mongoose.Schema;

var vendorSchema = new Schema({
    email : {type : String, required : true, unique : true},
    password : {type : String, required : true},
    vendorName : String,
    truckName : String,
    owner : String,
    description : String,
    location : String,
})

// export personSchema as a class called Person
module.exports = mongoose.model('Vendor', vendorSchema);

vendorSchema.methods.standardizeName = function() {
    this.name = this.name.toLowerCase();
    return this.name;
}